src/main/resources/testchipip/bootrom/bootrom.S

```diff
@@ -25,8 +25,13 @@ _hang: // reset vector
   csrw mtvec, a0
   li a0, 8           // MIE or MSIP bit
   csrw mie, a0       // set only MSIP in MIE CSR
-  csrw mideleg, zero // no delegation
   csrs mstatus, a0   // set MIE in mstatus CSR
+  csrr a0, misa      // get MISA mask
+  srai a0, a0, 18    // get 'S' supervisor bit
+  andi a0, a0, 0x1
+  beqz a0, wfi_loop  // only zero mideleg if supported
+  csrw mideleg, zero // no delegation
```

Comment:
Yes, `mstatus.MIE` should be set last to avoid a race.  `mideleg` is not required to be reset to any particular value.